haystack/components/generators/azure.py

```diff
@@ -126,6 +132,7 @@ def __init__(
         self.organization = organization
         self.model: str = azure_deployment or "gpt-35-turbo"
         self.timeout = timeout
+        self.max_retries = max_retries
```

Comment:
You can do something like:
```
self.max_retries = max_retries or int(os.environ.get("OPENAI_MAX_RETRIES", 5))
```
and avoid the code above (loc 119-121)

Reply:
Sounds good. Thanks for the suggestion 😄